asset-swapper: Stop requiring takerAddress for indicative RFQ-T quotes.

feuGeneA opened the pull request:
## Description

Stop requiring takerAddress for indicative RFQ-T quotes.

<!--- Describe your changes in detail -->

## Testing instructions

Verified via automated tests over in https://github.com/0xProject/0x-api/pull/320

## Types of changes

<!--- What types of changes does your code introduce? Uncomment all the bullets that apply: -->

<!-- * Bug fix (non-breaking change which fixes an issue) -->

* New feature (non-breaking change which adds functionality)

<!-- * Breaking change (fix or feature that would cause existing functionality to change) -->

feuGeneA committed:
asset-swapper: Stop req'ing takerAddr for RFQT ind

Stop requiring takerAddress for indicative RFQ-T quotes.

feuGeneA committed:
update CHANGELOG

feuGeneA committed:
Merge branch 'development' into feat/asset-swapper/rfqt-indicative-without-takerAddress

PirosB3 reviewed (changes requested):
When the QuoteRequestor makes a request to the RFQ client, they need to pass in a `takerAddress`. With these changes, do we risk an undefined `takerAddress`?

feuGeneA commented:
> When the QuoteRequestor makes a request to the RFQ client, they need to pass in a `takerAddress`. With these changes, do we risk an undefined `takerAddress`?

That's a great question @PirosB3 , thank you for asking it.  Thinking further, I think what we need to do is modify the `@0x/quote-server` to allow indicative quotes to skip passing a `takerAddress`.  I'll put up a PR for that.

feuGeneA committed:
asset-swapper: ensure takerAddress for indiciative

Ensure the presence of a taker address query parameter in the request to
market maker endpoints for indicative RFQ-T quotes.

Addresses review comment https://github.com/0xProject/0x-monorepo/pull/2684#pullrequestreview-477242367

feuGeneA committed:
Merge branch 'development' into feat/asset-swapper/rfqt-indicative-without-takerAddress

PirosB3 commented on packages/asset-swapper/src/utils/quote_requestor.ts:
```diff
@@ -217,7 +213,10 @@ export class QuoteRequestor {
         const _opts: RfqtRequestOpts = { ...constants.DEFAULT_RFQT_REQUEST_OPTS, ...options };
-        assertTakerAddressOrThrow(_opts.takerAddress);
+
+        if (!_opts.takerAddress) {
```
Would you mind adding a quick comment explaining why we set indicative quote `takerAddress` to `NULL_ADDRESS`?

feuGeneA replied:
Addressed in https://github.com/0xProject/0x-monorepo/pull/2684/commits/a8f3a47240c1a0adabbdd60ca82c8c98b4f07851

feuGeneA committed:
asset-swapper: Update comments
